experimental/iterators/include/iterators/Dialect/Iterators/IR/IteratorsTypes.td

```diff
@@ -36,29 +36,73 @@ class Iterators_Type<string name, string typeMnemonic, list<Trait> traits = []>
 //     well.
 //===----------------------------------------------------------------------===//
 
+/// List of printable float types.
+def Iterators_PrintableFloatType :
+  FloatOfWidths<[16, 32, 64]>;
+
+/// List of bit-widths of printable integer types.
+def Iterators_PrintableIntegerBitWidths {
+  list<int> widths = [ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10,
```

Comment:
there should be a way to get a confined CPred in tablgen

Reply:
Yes, good idea. Fixed in new commit.